### PR TITLE
feat(width): Introduce width utility

### PR DESCRIPTION
**Pull request by andybraren**
This adds a new width utility that enables breakpoint-specific width adjustments necessary for #785 and likely other responsive layouts in the future. We should probably discuss whether this is the right approach.

EDIT: This now includes `height` as well, making it more of a "Sizing" utility.

### CODE REVIEW

File: src/patternfly/utilities/Width/width.scss [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+/* stylelint-disable */
+
+// Width options
+$pf-u-width-options: (
+  width-auto:    (width auto),
+  width-full:    (width 100%),
+  width-initial: (width initial),
+  width-inherit: (width inherit)
+);
```
Comment (matthewcarleton) [inline src/patternfly/utilities/Width/width.scss]: should we add a width: 0; ?

File: src/patternfly/utilities/Width/width.scss [hunk 2]
```diff
@@ -0,0 +1,15 @@
+@import "../../patternfly-utilities";
+
+/* stylelint-disable */
+
+// Width options
+$pf-u-width-options: (
+  width-auto:    (width auto),
+  width-full:    (width 100%),
+  width-initial: (width initial),
```
Comment (matthewcarleton) [inline src/patternfly/utilities/Width/width.scss]: This is coming out as width: auto; @mattnolting / @michael-coker any thoughts on that?

Comment (michael-coker) [reply src/patternfly/utilities/Width/width.scss]: the initial value of width is `auto` so I don't think that's necessary. https://developer.mozilla.org/en-US/docs/Web/CSS/width

File: src/patternfly/utilities/Sizing/examples/sizing-button-example.hbs [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{{#> button button--modifier="pf-m-primary pf-m-block pf-u-w-initial-on-md"}}
```
Comment (mcoker) [inline src/patternfly/utilities/Sizing/examples/sizing-button-example.hbs]: this button example has a dashed border, probably due to some workspace scss

File: src/patternfly/utilities/Sizing/examples/sizing-breakpoint-example.hbs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{{#> display display--type="flex" display--modifier="pf-u-flex-direction-column"}}
```
Comment (mcoker) [inline src/patternfly/utilities/Sizing/examples/sizing-breakpoint-example.hbs]: should we add examples using `.pf-u-h-*`?

File: src/patternfly/utilities/Sizing/sizing.hbs [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<div class="item{{#if sizing--modifier}} {{sizing--modifier}}{{/if}}"
+  {{#if sizing--attribute}}
+    {{{sizing--attribute}}}
+	{{/if}}>
```
Comment (mcoker) [inline src/patternfly/utilities/Sizing/sizing.hbs]: Can you fix the indentation in this file?

File: src/patternfly/utilities/Width/width.scss (hunk 2 above)
Comment (andybraren) [reply src/patternfly/utilities/Width/width.scss]: `auto` is now removed. I've added a parenthetical "(auto)" to each example to point out that `-initial` is effectively the same as `auto`.

File: src/patternfly/utilities/Sizing/examples/sizing-button-example.hbs (hunk 3 above)
Comment (andybraren) [reply src/patternfly/utilities/Sizing/examples/sizing-button-example.hbs]: I added some CSS to the workspace to specifically target this one example. I suppose this button demo could be removed too, although it's a good illustration of its value.

File: src/patternfly/utilities/Sizing/sizing.hbs (hunk 5 above)
Comment (andybraren) [reply src/patternfly/utilities/Sizing/sizing.hbs]: Must've been some weird newline stuff that VS Code was correcting automatically for me. Fixed.

File: src/patternfly/utilities/Sizing/examples/sizing-breakpoint-example.hbs (hunk 4 above)
Comment (andybraren) [reply src/patternfly/utilities/Sizing/examples/sizing-breakpoint-example.hbs]: Done, thanks for the help!

**Comment by matthewcarleton** [comment]
Hey Andy, this looks great. Thanks for jumping back into it. A few comments: 
1. Should we include demos here? We don't in any of the other utilities so I'm not sure it's something we want to introduce.
2. Instead of adding a class to the preview I believe there is a `minHeight` parameter you can use.
3. It is a little odd to not include the zero value for both height and width. It might be missed if we don't follow the other examples. I see how it is a little odd though. @mcoker @srambach what do you think?

**Comment by mcoker** [comment]
+1 for zero value. And agreed, the demo is nice but doesn't belong.

**Comment by matthewcarleton** [comment]
Ya, and @mcoker pointed out the minheight won't work here. So let's stick with what you have. I'll create an issue for adding `height` to the workspace.

**Pull request merged**